libraries/AP_Scripting/examples/winch-test.lua

```diff
@@ -0,0 +1,102 @@
+-- mount-test.lua: allows the winch to be deployed or retracted at a fixed speed using an auxiliary switch
+--
+-- How To Use
+--   1. set RCx_OPTION to 300 to enable controlling the winch rate from an auxiliary switch
+--   2. optionally set WINCH_RATE_UP to the fixed retract speed (in m/s)
+--   3. optionally set WINCH_RATE_DN to the fixed deploy speed (in m/s)
+--   4. raise the RC auxiliary switch to retract the winch's line
+--   5. lower the RC auxiliary switch to deploy the winch's line
+--   6. center the RC auxiliary switch to stop the winch
+-- Alternatively a servo *output* can be used in place of the auxiliary switch input by setting WINCH_SRV_SRC_FN to match a servo channel's function.  For example
+--   a. set SERVO10_FUNCTION = 28 (Gripper)
+--   b. set WINCH_SRV_SRC_FN to 28
+--   c. use Mission Planner's Data screen's Servo/Relay tab to set the SERVO10 output to Low, Mid or High values
+--   Note: the full list of SERVOx_FUNCTION values that will work are 0:None, 1:Manual, 22:SprayerPump, 23:SprayerSpinner, 28/Gripper, 51:RCIN1 to 66:RCIN16
+--
+
+-- global definitions
+local UPDATE_INTERVAL_MS = 100
+
+-- add new parameters
+local PARAM_TABLE_KEY = 80
+assert(param:add_table(PARAM_TABLE_KEY, "WINCH_", 3), "could not add param table")
+assert(param:add_param(PARAM_TABLE_KEY, 1, "RATE_UP", 0.5), "could not add WINCH_RATE_UP param")
+assert(param:add_param(PARAM_TABLE_KEY, 2, "RATE_DN", 2), "could not add WINCH_RATE_DN param")
+assert(param:add_param(PARAM_TABLE_KEY, 3, "SRV_SRC_FN", -1), "could not add WINCH_SRV_SRC_FN param")
+
+local winch_rate_up = Parameter()
+assert(winch_rate_up:init("WINCH_RATE_UP"), "could not find WINCH_RATE_UP param")
+
+local winch_rate_dn = Parameter()
+assert(winch_rate_dn:init("WINCH_RATE_DN"), "could not find WINCH_RATE_DN param")
+
+local winch_srv_src_fn = Parameter()
+assert(winch_srv_src_fn:init("WINCH_SRV_SRC_FN"), "could not find WINCH_SRV_SRC_FN param")
```

Comment:
On master you can just:
```suggestion
local winch_rate_up = Parameter"WINCH_RATE_UP"()
local winch_rate_dn = Parameter("WINCH_RATE_DN")
local winch_srv_src_fn = Parameter("WINCH_SRV_SRC_FN")
```

Reply:
cool, txs.  fixed.